libfwupd/fwupd-common.c

```diff
@@ -330,7 +346,7 @@ fwupd_build_user_agent (const gchar *package_name, const gchar *package_version)
 
 /**
  * fwupd_build_machine_id:
- * @salt: The salt, or %NULL for none
+ * @fn: The path of machine-id
```

Comment:
this is a _weak_ ABI change.  The function is exported to all clients to be able to potentially use.  Even though the data type didn't change, the behavior of it changed.  The filename is now used for the salt, and it actually is required.

I think we actually need a new symbol for this new function behavior and to deprecate the old one with the next ABI bump.